lib/js-file.js

```diff
@@ -542,7 +554,238 @@ JsFile.prototype = {
         });
 
         return lines;
+    },
+
+    /**
+     * Renders JS-file sources using token list.
+     *
+     * @returns {String}
+     */
+    render: function() {
+        var result = '';
+
+        // For-loop for maximal speed.
+        for (var i = 0; i < this._tokens.length; i++) {
+            var token = this._tokens[i];
+
+            result += token.whitespaceBefore;
```

Comment:
http://jsperf.com/string-concat-vs-array-join-10000/15 suggests that this doesn't matter.